test/ApplicationTest.php

```diff
@@ -27,7 +29,22 @@
 
 class ApplicationTest extends TestCase
 {
-    public function setUp()
+    /** @var MiddlewareFactory|ObjectProphecy */
+    private $factory;
+
+    /** @var MiddlewarePipeInterface|ObjectProphecy */
+    private $pipeline;
+
+    /** @var RouteMiddleware|ObjectProphecy */
+    private $routes;
+
+    /** @var RequestHandlerRunner|ObjectProphecy */
+    private $runner;
+
+    /** @var Application */
+    private $app;
```

Comment:
Are private properties needed in Test classes?

Reply:
It doesn't matter, it was not done automatically, I've added them manually where I've noted there are missing. These could be `protected` or even `public`, I just think that it's nice to have only `public test*` method and all others `private`. We don't need to have even these properties and we can define them dinalically but you know - for IDE it's better to have them in place :)

Reply:
This is one area where I'm definitely on the fence. Test classes do not have the same context as the source code for the package; I will never, ever use phpDoc on them, and appreciate the ability to create new properties on an ad-hoc basis.

That said, I totally understand folks who use IDEs and want to have the typehints, so they know what they're allowed to call, and with what arguments.

Again, I think we need a poll on this to see if there's consensus with a broader audience.